### DISCUSSION TITLE
LFX Mentorship 2023 03-Sep-Nov Challenge - for #2680

### DISCUSSION THREAD

**Discussion by dannypsnl** [WasmEdge/WasmEdge#2681] [Show and tell]
# Introduction

For those who want to apply for LFX mentorship for #2680, this is a pretest for the application. The goal of this LFX mentorship is to extend plugin OpenCVMini for WasmEdge. We designed this challenge to evaluate the candidates.

# Requirements

### First part: Build WasmEdge with plugins from source code 

This repository, which means you should only dealing with `cmake` and `ninja`

### Second part: you will create a repository which contains

1. A C++ program that using OpenCV to do image filtering.
2. Once you are finished, please provide your link to the GitHub repository as a comment under this discussion, which must contain a README file to show how you do it.

# Deadline

Please refer to [the timeline of LFX mentorship 2023 03-Sep-Nov](https://github.com/cncf/mentoring/tree/main/programs/lfx-mentorship/2023/03-Sep-Nov).
Since we have to process the internal review and decide, the final date for valid submissions will be Aug 27, 5:00 PM PDT.

**Comment by PurviChaurasia** [comment]
Hi @dannypsnl, I was facing some issue while setting up and building WasmEdge. While running the cmake command `cmake -Bbuild -GNinja -DCMAKE_SYSTEM_VERSION=10.0.19041.0 -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDLL "-DLLVM_DIR=$llvm_dir" -DWASMEDGE_BUILD_TESTS=ON -DWASMEDGE_BUILD_PACKAGE="ZIP" .` I am getting the following error - 
` CMake Warning:
  Ignoring extra path from command line:

   ".0.19041.0"


CMake Error at C:/Program Files/CMake/share/cmake-3.27/Modules/CMakeDetermineCCompiler.cmake:49 (message):
  Could not find compiler set in environment variable CC:

  clang-cl.
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred! `

I tried searching on the web and did find a relevant [stackoverflow post](https://stackoverflow.com/questions/70524164/cmake-c-compiler-not-set-after-enablelanguage) however, am not sure what to add as path. Could you please help out with this?

**Reply by hydai** [reply]
Wrap the versions with `"10.0.19041.0"`.
Please install the compilers, or set the $CC env variable.

**Comment by vrnimje** [comment]
Hi @dannypsnl , I have successfully built and installed WasmEdge with plugins on my system, using `cmake` & `ninja`
![image](https://github.com/WasmEdge/WasmEdge/assets/103848930/ae96f46b-4704-4372-94bd-dcb632d2472d)

And here is the repository, where I used OpenCV to apply some filters to images: [opencv-filters](https://github.com/vrnimje/opencv-filters.git)

**Comment by 1awesomeJ** [comment]
@dannypsnl, good evening, I'm Josh from Nigeria, I'm an aspiring mentee.
I have cloned the repository, built and run the test, with all 24 passing.
Do we need to show any proof of that first task done, or is it just for the second task that we need to drop a link?
Thank you!

**Reply by dannypsnl** [reply]
A picture is enough, anyway mentee will do this build all the time in this mentorship, cheating won’t help

**Comment by 1awesomeJ** [comment]
@dannypsnl, here's a screenshot of my build:
<img width="960" alt="wasmedge_with_plugins" src="https://github.com/WasmEdge/WasmEdge/assets/109356020/c126e17f-7e57-40cd-869b-5f8445e7bc7c">

**Comment by 1awesomeJ** [comment]
@dannypsnl,
here's the link to my repository of a C++ program that uses OpenCV to do image filtering(The README included):
https://github.com/1awesomeJ/WasmEdge_OpenCV

**Comment by PurviChaurasia** [comment]
Hi @hydai I have successfully completed the pre-test. Here is the screenshot of the plugins setup from source for WasmEdge.
![image](https://github.com/WasmEdge/WasmEdge/assets/97350598/fa12529c-db33-47af-9be4-9c64c143a849)
I also also created a repository for the image filtering using OpenCV in C++ - [Image Filtering](https://github.com/PurviChaurasia/OpenCV-ImageFilters-WasmEdge/tree/main)
Kindly let me know if there any changes or modifications that have to be made. I am open to all suggestions and guidance!

**Comment by Mash707** [comment]
Hii @dannypsnl , I have successfully built Wasmedge from source code with plugins. Here is a screenshot showing all plugins.
![Plugins](https://github.com/WasmEdge/WasmEdge/assets/114072061/18a6e30e-f630-4d97-88ae-acfa9d9a3ce6)
I have also created a repository which contains a [Image-Filtering](https://github.com/Mash707/OpenCV-Image-Filtering) C++ program using OpenCV. 
Please let me know if any changes should be done.

**Comment by aaryac33** [comment]
https://github.com/aaryac33/Add-matrix-operations-for-Open-CV-Mini-Wasm-Plugin

I am really interested in working on this project. Above is my pretest submission.

**Comment by alabulei1** [comment]
Hi @PurviChaurasia, @aaryac33, @vrnimje, @1awesomeJ, and @Mash707,

Thank you for completing the pretest. If you have not yet submitted your application for this project, please do so before 5 PM PDT on August 25th. 

Apply here: https://mentorship.lfx.linuxfoundation.org/project/f21bec94-085e-4e2d-94ac-3b61b4471818

**Comment by macroexpansion** [comment]
<img width="885" alt="Screenshot 2023-08-26 at 11 56 44" src="https://github.com/WasmEdge/WasmEdge/assets/25638688/3c361be3-9843-4038-9c00-108c63cf7081">

Hi, this is my C++ program for image filtering [Image Filter](https://github.com/blueyellowpink/wasmedge-image-filter)

Best regards,
Andrew

**Comment by Wck-iipi** [comment]
![proof](https://github.com/WasmEdge/WasmEdge/assets/110763795/d5f467bf-9a71-4061-a80c-7333d6b726e7)

Hello, I have uploaded the proof of installation of plugins and have created a [repository](https://github.com/Wck-iipi/LFX_OpenCV) implementing 16 different [image filtering functions](https://docs.opencv.org/4.x/d4/d86/group__imgproc__filter.html).
I am interested in the project and would love to work on WasmEdge.

**Comment by xevor11** [comment]
Unable to load the build due to technical issues
![Screenshot 2023-08-30 020447](https://github.com/WasmEdge/WasmEdge/assets/74680679/bd0591ab-5537-4cd8-844b-04d59bc9ed2e)
